plugin/restore/mysql/mysql.go

```diff
@@ -123,6 +132,170 @@ func GetPITRDatabaseName(database string, timestamp int64) string {
 	return getSafeName(database, suffix)
 }
 
+// SyncBinlogs syncs the binlogs between the instance and `saveDir`, but exclude latest binlog
+func (r *Restore) SyncBinlogs(ctx context.Context, instance *api.Instance, saveDir string) error {
+	// 1. read the binlog file in saveDir
+	binlogFilesLocal, err := ioutil.ReadDir(saveDir)
+	if err != nil {
+		return err
+	}
+
+	// 2. get the metadata of binlogs on the server
+	binlogFilesOnServer, err := r.showBinlogFiles(ctx)
+	if err != nil {
+		return err
+	}
+
+	// 3. get meta of the latest binlog
+	latest, err := r.showLatestBinlogFile(ctx)
+	if err != nil {
+		return err
+	}
+
+	// 4. compare file sizes and name to decide which files to download
+	downloadedIndex := make(map[int]struct{})
+	for index, serverFile := range binlogFilesOnServer {
+		// We don't download the latest binlog in SyncBinlogs()
+		if serverFile.Name == latest.Name {
+			continue
+		}
```

Comment:
How about just removing the last element from the result of "SHOW MASTER STATUS", and the list will only contains archived binlog files.

Reply:
> How about just removing the last element from the result of "SHOW MASTER STATUS", and the list will only contain archived binlog files.

I'm not sure if mysql will guarantee return in order

Reply:
SG.